plugins/rating/services/RatingService.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * Allows user to manipulate their entry rating
+ *
+ * @service rating
+ * @package plugins.rating
+ * @subpackage api.services
+ */
+
+class RatingService extends KalturaBaseService
+{
+	public function initService($serviceId, $serviceName, $actionName)
+	{
+		parent::initService($serviceId, $serviceName, $actionName);
+		
+		if(!RatingPlugin::isAllowedPartner($this->getPartnerId()))
+		{
+			throw new KalturaAPIException(KalturaErrors::FEATURE_FORBIDDEN, RatingPlugin::PLUGIN_NAME);
+		}
+		
+		if ((!kCurrentContext::$ks_uid || kCurrentContext::$ks_uid == "") && $actionName != "getRatingCounts")
+		{
+			throw new KalturaAPIException(KalturaErrors::INVALID_USER_ID);
+		}
+	}
+	
+	/**
+	 * @action rate
+	 * Action for current kuser ro rate a specific entry
+	 * @param string $entryId
+	 * @param int $rank
+	 * @throws KalturaErrors::ENTRY_ID_NOT_FOUND
+	 * @throws KalturaErrors::INVALID_RANK_VALUE
+	 * @return int
+	 */
+	public function rateAction ($entryId, $rank)
+	{
+		if (!$entryId)
```

Comment:
This check is not needed since entry has no default value , the dispatcher will validate that you have an string value in entry